langkit/lkt_lowering.py

```diff
@@ -47,13 +47,17 @@
 ANNOTATIONS_WHITELIST = ['builtin']
 
 
-def get_trait(decl: L.TypeDecl, trait_name: str) -> Optional[L.TypeDecl]:
+def get_trait(decl: L.TypeDecl,
+              trait_name: str,
+              by_syn_name: bool = True) -> Optional[L.TypeDecl]:
     """
     Return the trait named ``trait_name`` on declaration ``decl``.
     """
     for trait in decl.f_traits:
         trait_decl: L.TypeDecl = trait.p_designated_type
-        if trait_decl.f_syn_name.text == trait_name:
+        if by_syn_name and trait_decl.f_syn_name.text == trait_name:
+            return trait_decl
+        elif trait_decl.p_name == trait_name:
```

Comment:
I don’t understand why this hunk is necessary: the only case where `by_syn_name` is False is for the `Node` trait, and `f_syn_name.text` should always return "None" in this case. Could we remove this part of the patch?

Reply:
As discussed live, this (and the `by_syn_name` param) are unneeded AFAICT, you can just remove this branch and the param

Reply:
OK, thanks for the explanation!